Base/QTGUI/qSlicerSettingsViewsPanel.cxx

```diff
@@ -296,7 +296,7 @@ void qSlicerSettingsViewsPanel::sliceViewOrientationChangedByUser()
       {
       // Non-default orientation is chosen and no orientation marker is displayed.
       // To ensure that there is no accidental mixup of orientations, show the orientation marker.
-      d->SliceOrientationMarkerTypeComboBox->setCurrentText("axes");
+      d->SliceOrientationMarkerTypeComboBox->setCurrentText(/*no tr*/"axes");
```

Comment:
Combobox can store user data for each item. It would be nicer to translate the item text and store the user data (the nob-translated text) in application settings and ithe scene.

There is a mapper that allows registering a settings property with userdata. If you don't find it then let me know.

Reply:
I understand the problem you raised but I don't find the mapper you are talking about.

I wanted to rewrite the code as follows but I don't have the guarantee that the "None" item's data was initialised to "none". 

```cpp
if (d->SliceOrientationMarkerTypeComboBox->currentData() == "none")
{
  // Non-default orientation is chosen and no orientation marker is displayed.
  // To ensure that there is no accidental mixup of orientations, show the orientation marker.
  int index = d->SliceOrientationMarkerTypeComboBox->currentIndex();
  d->SliceOrientationMarkerTypeComboBox->setItemData(index, /*no tr*/"axes");
  d->SliceOrientationMarkerTypeComboBox->setCurrentText(tr("axes"));
}
```

So I finally mark the current text as translatable (both `none` and `axes`). I planned to update it whenever I have more information on that mapper.

Reply:
The combobox items should be removed from the .ui file and instead the items (translated text and non-translated userdata) should be added as it is done here (just using qSlicerSettingsViewsPanel::tr instead of QWidget::tr):

https://github.com/Slicer/Slicer/blob/6c67bc6b8f637c5c31ec5aadf404778ce774f4bb/Base/QTGUI/qSlicerSettingsViewsPanel.cxx#L105-L112

Reply:
Ok, it's done now.

By the way, which of the following approach is the best when it's about dealing with commits related to PR comments :

- Add new commits at each update that concerns one or more comments ?
- Always merge all commits and force push the branch  so that there is always one commit related to the PR ?

Thank you !

Reply:
Thank you!

There is no strict rule for additional commits or force push. When there are merge conflicts or things just get too complicated then it makes sense to do a force-push. Otherwise additional commits are nice because they don't invalidate the comments.

Reply:
Ok, that's good then.

Thank you again for all of your valuable feedback!